internal/webhooks/clusterclass.go

```diff
@@ -178,9 +175,10 @@ func (webhook *ClusterClass) validateRemovedMachineDeploymentClassesAreNotUsed(c
 		for _, machineDeploymentTopology := range c.Spec.Topology.Workers.MachineDeployments {
 			if removedClasses.Has(machineDeploymentTopology.Class) {
 				// TODO(killianmuldoon): Improve error printing here so large scale changes don't flood the error log e.g. deduplication, only example usages given.
+				// TODO: consider if we get the index of the MachineDeploymentClass being deleted
```

Comment:
I think that depends on if the path should apply to the old or new resource (or depending on the case one or the other)